sycl/unittests/xpti_trace/QueueApiFailures.cpp

```diff
@@ -0,0 +1,502 @@
+protected:
+  void SetUp() {
+    xptiForceSetTraceEnabled(true);
+    xptiTraceTryToEnable();
+  }
+
+  void TearDown() {
+    resetReceivedNotifications();
+    xptiForceSetTraceEnabled(false);
+  }
+
+public:
+  static std::string BuildCodeLocationMessage(const char *FileName,
+                                              const char *FunctionName,
+                                              int LineNumber,
+                                              int ColumnNumber) {
+    const char Delimiter[] = ";";
+    std::string Result(FunctionName);
+    Result.append(Delimiter);
+    Result.append(FileName);
+    Result.append(Delimiter);
+    Result.append(std::to_string(LineNumber) + Delimiter +
+                  std::to_string(ColumnNumber));
+    return Result;
+  }
+
+  unittest::ScopedEnvVar PathToXPTIFW{"XPTI_FRAMEWORK_DISPATCHER",
+                                      "libxptifw.so", [] {}};
+  unittest::ScopedEnvVar XPTISubscriber{"XPTI_SUBSCRIBERS",
+                                        "libxptitest_subscriber.so", [] {}};
+  sycl::unittest::PiMock MockPlugin;
+
+  static constexpr char FileName[] = "QueueApiFailures.cpp";
+  static constexpr char FunctionName[] = "TestCaseExecution";
+  static constexpr char ExtraFunctionName[] = "ExtraTestCaseExecution";
+  static constexpr int LineNumber = 8;
+  static constexpr int ExtraLineNumber = 7;
+  static constexpr int ColumnNumber = 13;
+  const sycl::detail::code_location TestCodeLocation = {
+      FileName, FunctionName, LineNumber, ColumnNumber};
+  const sycl::detail::code_location ExtraTestCodeLocation = {
+      FileName, ExtraFunctionName, ExtraLineNumber, ColumnNumber};
+  static constexpr size_t KernelSize = 1;
+  using TestKI = detail::KernelInfo<TestKernel<KernelSize>>;
+
+  const std::string TestCodeLocationMessage = BuildCodeLocationMessage(
+      FileName, FunctionName, LineNumber, ColumnNumber);
+  const std::string ExtraTestCodeLocationMessage = BuildCodeLocationMessage(
+      FileName, ExtraFunctionName, ExtraLineNumber, ColumnNumber);
+  const std::string TestKernelLocationMessage = BuildCodeLocationMessage(
+      TestKI::getFileName(), TestKI::getFunctionName(), TestKI::getLineNumber(),
+      TestKI::getColumnNumber());
+  const std::string PiLevelFailMessage = "Native API failed";
+  const std::string UnknownCodeLocation = "code location unknown";
+};
+
+TEST_F(QueueApiFailures, QueueSubmit) {
+  MockPlugin.redefine<detail::PiApiKind::piEnqueueKernelLaunch>(
+      redefinedEnqueueKernelLaunch);
+  MockPlugin.redefine<detail::PiApiKind::piPluginGetLastError>(
+      redefinedPluginGetLastError);
+  sycl::queue Q;
+  bool ExceptionCaught = false;
+  try {
+    Q.submit(
+        [&](handler &Cgh) {
+          Cgh.single_task<TestKernel<KernelSize>>([=]() {});
+        },
+        TestCodeLocation);
+  } catch (sycl::exception &e) {
+    ExceptionCaught = true;
+  }
+  Q.wait();
+  EXPECT_TRUE(ExceptionCaught);
```

Comment:
Should this be an `ASSERT_TRUE` instead? I figure there won't be much point in doing the following checks if the expected error wasn't caught.

Reply:
thanks, done